nixos/modules/services/web-servers/traefik.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.traefik;
+  configFile = pkgs.writeText "traefik.toml" cfg.config;
+in {
+  options.services.traefik = {
+    enable = mkEnableOption "Traefik web server";
+
+    config = mkOption {
+      default = "";
+      example = ''
+      defaultEntryPoints = ["http"]
+
+      [entryPoints]
+      [entryPoints.http]
+      address = ":80"
+
+      [web]
+      address = ":8080"
+      '';
+      type = types.lines;
+      description = "Verbatim traefik.toml to use";
+    };
+
+    dataDir = mkOption {
+      default = "/var/lib/traefik";
+      type = types.path;
+      description = ''
+      Location for any persistent data traefik creates, ie. acme
+      '';
+    };
+
+    package = mkOption {
+      default = pkgs.traefik;
+      defaultText = "pkgs.traefik";
+      type = types.package;
+      description = "Traefik package to use.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.traefik = {
+      description = "Traefik web server";
+      after = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = ''${cfg.package.bin}/bin/traefik -l debug --configfile=${configFile}'';
```

Comment:
Is debug a good default for logging?

Reply:
Ah sorry, I forgot to remove that. I had it there while debugging.